tools/sgmdformat/tools.go

```diff
@@ -60,3 +63,26 @@ func PrepareCommand(ctx context.Context) error {
 	}
 	return nil
 }
+
+// list markdown files known by git + untracked ones.
+func listMarkdownFiles(ctx context.Context) []string {
+	output := strings.TrimSpace(
+		sg.Output(sggit.Command(
+			ctx,
+			"ls-files",
+			"*.md",               // only markdown files
+			"--others",           // include untracked files
+			"--exclude-standard", // exclude ignored files
+			"--cached",           // include "normal" files
```

Comment:
So this excludes any .gitignored files and allows everything else?

Is `--cached` needed?

Reply:
Yes, cached is the default flag if you don't invoke e.g. --others. So since we use --others, then we also need to include --cached to get all files, not just untracked ones.